.github/workflows/firebase-hosting-pull-request.yml

```diff
@@ -2,13 +2,16 @@
 # https://github.com/firebase/firebase-tools
 
 name: Deploy on PR
-on: [pull_request_target]
+on:
+    pull_request_target:
+        types: [labeled]
 
 permissions: read-all
 
 jobs:
     build_and_preview:
         runs-on: ubuntu-latest
+        if: contains(github.event.pull_request.labels.*.name, 'safe to test')
```

Comment:
Let's name it more friendly. Since it's only about deployments, how about we add a label called "Deploy"?